include/courtroom.h

```diff
@@ -258,14 +258,10 @@ class Courtroom : public QMainWindow
     ERR_CREATELOG, //couldn't create log file
     SUCCESS
   };
-  save_log_result ic_result;
-  save_log_result ooc_result;
-
-  // Attempts to save the IC log to a file, and sets ic_result to the result. 
-  // Returns the name of the log file (even if the file was not created)
-  QString save_ic_chatlog();
-  // Same thing, but for the OOC log.
-  QString save_ooc_chatlog();
+
+  // Attempts to save a chatlog to a file, returns the result as an integer corresponding to
+  // the above enumeration. is_ooc tells it which log to save - if true, save OOC, if false, save IC.
+  int save_chatlog(bool is_ooc);
```

Comment:
You have nothing to lose if you use an enum for your error code:
```suggestion
  save_log_result save_chatlog(bool is_ooc);
```

Reply:
I was almost certain this wasn't possible, but as it turns out I was neglecting to add `Courtroom::` to the function type in `text_file_functions.cpp`. Live and learn, I guess.